### PR TITLE
Patches:

### PR DESCRIPTION
**Pull request by Isfirs**
- Added a MessageMode, determined by the cosntructor used.
- Changed server method properly
- Changed some result types in ItermPermissionScepter

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/network/messages/ChangeFreeToInteractBlockMessage.java [hunk 1]
```diff
@@ -119,29 +135,40 @@ public void messageOnServerThread(final ChangeFreeToInteractBlockMessage message
...
-                {
-                    colony.addFreeBlock(message.block);
-                }
-                else
+                switch (message.mode)
                 {
-                    colony.addFreePosition(message.pos);
+                    case LOCATION:
+                        colony.addFreePosition(message.pos);
+                        LanguageHandler.sendPlayerMessage(player, "com.minecolonies.coremod.item.permissionscepter.addposition.success", message.pos.getX(),message.pos.getY(),message.pos.getZ());
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/network/messages/ChangeFreeToInteractBlockMessage.java]: MAJOR Split this 195 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/network/messages/ChangeFreeToInteractBlockMessage.java [hunk 2]
```diff
@@ -119,29 +135,40 @@ public void messageOnServerThread(final ChangeFreeToInteractBlockMessage message
...
-                {
-                    colony.removeFreeBlock(message.block);
-                }
-                else
+                switch (message.mode)
                 {
-                    colony.removeFreePosition(message.pos);
+                    case LOCATION:
+                        colony.removeFreePosition(message.pos);
+                        LanguageHandler.sendPlayerMessage(player, "com.minecolonies.coremod.item.permissionscepter.removelocation.success", message.pos.getX(),message.pos.getY(),message.pos.getZ());
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/network/messages/ChangeFreeToInteractBlockMessage.java]: MAJOR Split this 198 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 43 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 41 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [jobs](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/coremod/colony/jobs): Add a 'package-info.java' file to document the 'jobs' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/2998db777d79b979cdf06a3ce7a7d6275051df1b/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 41 issues
* ![MINOR][MINOR] 41 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [jobs](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/coremod/colony/jobs): Add a 'package-info.java' file to document the 'jobs' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/cf3d7016238cc3f6447d99c87890f9a7ee22c8dd/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]
Approved.

**Pull request merged**